Loop/Managers/Alerts/StoredAlert.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  StoredAlert.swift
+//  Loop
+//
+//  Created by Rick Pasetto on 5/11/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import CoreData
+import LoopKit
+
+extension StoredAlert {
+
+    static var encoder = JSONEncoder()
+          
+    convenience init(from deviceAlert: DeviceAlert, context: NSManagedObjectContext, issuedTimestamp: Date = Date()) {
+        do {
+            self.init(context: context)
+            self.issuedTimestamp = issuedTimestamp
+            identifier = deviceAlert.identifier.value
+            // Encode as JSON strings
+            let encoder = StoredAlert.encoder
+            trigger = try encoder.encodeToStringIfPresent(deviceAlert.trigger)
+            sound = try encoder.encodeToStringIfPresent(deviceAlert.sound)
+            foregroundContent = try encoder.encodeToStringIfPresent(deviceAlert.foregroundContent)
+            backgroundContent = try encoder.encodeToStringIfPresent(deviceAlert.backgroundContent)
+            isCritical = deviceAlert.foregroundContent?.isCritical ?? false || deviceAlert.backgroundContent?.isCritical ?? false
+            modificationCounter = 0
```

Comment:
[nit] The other Core Data models do not explicitly initialize `modificationCounter` in code (because I didn't want folks thinking that it was something that should be manually modified. I believe they all configure the property in the model file to default to 0.

Reply:
Up to you, though.

Reply:
👍 Nice.  I'd rather do it that way.